tools/mpremote/mpremote/main.py

```diff
@@ -317,31 +319,38 @@ def argparse_none(description):
     "--version": "version",
 }
 
-# Add "a0", "a1", ..., "u0", "u1", ..., "c0", "c1", ... as aliases
+# Add "a0", "a1", ..., "u0", "u1", ..., "c1", "c2", ... as aliases
 # for "connect /dev/ttyACMn" (and /dev/ttyUSBn, COMn) etc.
 for port_num in range(4):
     for prefix, port in [("a", "/dev/ttyACM"), ("u", "/dev/ttyUSB"), ("c", "COM")]:
-        _BUILTIN_COMMAND_EXPANSIONS["{}{}".format(prefix, port_num)] = {
-            "command": "connect {}{}".format(port, port_num),
-            "help": 'connect to serial port "{}{}"'.format(port, port_num),
+        if port_num == 0 and port == "COM":
+            continue  # skip COM0 as it does not exist
+        _BUILTIN_COMMAND_EXPANSIONS[f"{prefix}{port_num}"] = {
+            "command": f"connect {port}{port_num}",
+            "help": f'connect to serial port "{port}{port_num}"',
         }
 
 
 def load_user_config():
     # Create empty config object.
     config = __build_class__(lambda: None, "Config")()
     config.commands = {}
-
-    # Get config file name.
-    path = os.getenv("XDG_CONFIG_HOME")
-    if path is None:
-        path = os.getenv("HOME")
+    # use $XDG_CONFIG_HOME, if on Windows use $APPDATA
+    if os.name != "nt":
```

Comment:
For a typical MSYS2 installation with a mingw-w64-x86_64-python3 installed, which is used a lot with MicroPython I think, and is what the non-msvc Windows port is built for, this is not ideal because its name is `nt` but it has a `HOME` directory and that is what you'd want to be used. There might be ways to detect that, but simpler is to just use `APPDATA` as fall back only `if not getenv('HOME') and `os.name == nt`.

Reply:
Thanks for pointing that out. 
Its hard finding a 'weighted good' that works reliably and understandably across platforms. 
What are the changes of  such a 'MSYS2 installation with a mingw-w64-x86_64-python3' using "XDG_CONFIG_HOME" ?

Reply:
> using "XDG_CONFIG_HOME"

That's not a thing in standard MSYS2 as far as I can tell, i.e. if people want to use it they likely declare it themselves, but in any case: it would follow the same logic as there is now, so XDG_CONFIG_HOME taking precedence over HOME.

Reply:
was already the care - Confirmed via tests for the the current PR